src/rudymap/root_leaf.rs

```diff
@@ -17,6 +17,7 @@ pub trait RootLeaf<K: Key, V> {
     fn remove(&mut self, key: K) -> RemoveResult<V>;
     fn shrink_remove(self, key: K) -> (RootPtr<K, V>, V);
     fn len(&self) -> usize;
+    fn memory_usage(&self) -> usize;
```

Comment:
How about replacing this with a default method:

```rust
fn memory_usage(&self) -> usize {
    mem::size_of::<Self>()
}
```

That should work for all other non-JPM roots.